src/librustc_resolve/imports.rs

```diff
@@ -1467,7 +1467,9 @@ impl<'a, 'b> ImportResolver<'a, 'b> {
 
         if !reexports.is_empty() {
             if let Some(def_id) = module.def_id() {
-                self.r.export_map.insert(def_id, reexports);
+                // Call to `expect_local` should be fine because current
+                // code is only called for local modules.
+                self.r.export_map.insert(def_id.expect_local(), reexports);
```

Comment:
@petrochenkov this is the call to `expect_local` that I was afraid might break things but it seems to work.

Reply:
Yes, `finalize_resolutions` is all about the current crate.